tests/unit/test_estimator.py

```diff
@@ -990,6 +990,17 @@ def test_generic_deploy_accelerator_type(sagemaker_session):
     assert args[1][0]['InstanceType'] == INSTANCE_TYPE
 
 
+def test_deploy_with_update_endpoint(sagemaker_session):
+    e = Estimator(IMAGE_NAME, ROLE, INSTANCE_COUNT, INSTANCE_TYPE, output_path=OUTPUT_PATH,
+                  sagemaker_session=sagemaker_session)
+    e.set_hyperparameters(**HYPERPARAMS)
+    e.fit({'train': 's3://bucket/training-prefix'})
+    e.deploy(INSTANCE_COUNT, INSTANCE_TYPE, endpoint_name="endpoint-name", update_endpoint=True)
+    sagemaker_session.create_endpoint_config.assert_called()
+    sagemaker_session.update_endpoint.assert_called()
```

Comment:
Please use `assert_called_with` instead.

Reply:
Updated with using `call_args` and assert with argument values. The model_name/endpoint_config_name is not accessible from estimator, so not able to use `assert_called_with` directly.